oeps/processes/oep-0058-proc-translations-management.rst

```diff
@@ -0,0 +1,169 @@
+  makes the translation process more open source. 
+* The GitHub Transifex Integration App is developed and supported by Transifex
+* The Github Transifex App is very simple to configure and with many options. We can set
+  Transifex Projects to automatically upload/download translation files from a repository
+  once the translations are reviewed and accepted.
+* Transifex only allows a one-to-one relationship between repositories and Transifex
+  Projects. Organizing all of the translation files into one repository and one Transifex
+  Project has a lower cost: we pay by the project so we end up paying less, and by
+  putting all translation files in the same place, it is easier to track translation
+  progress, and debug translation issues.
+* A repo that only contains text/binary files, and uses branches to separate translations
+  related to Open edX releases can make all interactions with translations very quick and
+  simple due to the ability to clone the branch of a specific release, with translations
+  organized by repository name.
+* By using an app that is maintained by Transifex the organization, we reduce the
+  maintenance burden and are more future proof of changes they might make since they
+  maintain both the API and the github App.
+
+Consequences
+************
+
+Impact on Translators
+=====================
+
+As we approach the end of the translation upgrade process, we will need to tactically
+move from multiple transifex projects to a single project.  This will require
+coordination with our translators to ensure that moving forward they are providing
+translations in the right place.
+
+Impact on Site Operators
+========================
+
+Currently the translation files for any given service or library is stored at the same
+place as the code, this has generally simplified the deployment story in the past.  With
+this change, the translations files will move to their own repository.  As we deprecate
+the old translations files, the relevant deployment tooling will need to be updated to
+pull down the translations from the new repo as a part of the deployment process.  This
+will impact both the old Ansible based tooling as well as any new docker based tooling.
+
+Impact on Developers
+====================
+
+While it won’t directly impact the day-to-day workflow of developers, due to the same
+reasons that we impact site operators(new translations location), we will have to update
+development tools as well.
+
+Rejected Alternatives
+*********************
+
+Rewriting the Current Tooling for the New API
+=============================================
+
+While rewriting the current tooling may solve the problems encountered in the last two
+Open edX Releases, it will still be difficult to maintain and keep track of the
+edx-transifex-bot. The Open edX community will still struggle with debugging difficult
+issues that stem from missing source code.
```

Comment:
What evidence supports this?

Reply:
I'm not sure if there is hard evidence for this. 

At some point the current tooling worked. The edx-transifex-bot was still difficult to track its activity. Debugging Transifex/translations issues was very time consuming because of the missing source code and difficulty finding where changes the bot made were bad.

Even if the current tooling was upgraded to work with the new API version, it's still the same tooling, and so this won't solve all the other issues unless we rewrite more of the tooling. But this is all actually a moot point, without access to the source code, how can we update the source code?

Reply:
Ah. I would tighten this and make it stronger. Maybe,

"The source code for the edx-transifex-bot is missing. We could rewrite the current tooling to try to solve the problems encountered in the last two Open edX releases and upgrade to the new API, but this approach would require a full rewrite, potentially more expensive than doing the rewrite in a way that Transifex more cleanly supports. It should also be mentioned that GitHub discourages the use of bots and separate bot accounts; they strongly recommend using GitHub Apps."

Your paragraph doesn't lead with the simple fact the source is missing. I'm not sure what you mean by "it will be difficult to maintain and keep track of the bot". That doesn't seem like a very strong argument - if you rewrite the bot, it doesn't connect for me that maintaining the bot will be difficult. Stronger arguments are - source is missing, we have to do a full rewrite, and we'd be doing a rewrite in a way that Transifex & GitHub do not recommend.